src/Controller/HomeController.php

```diff
@@ -4,16 +4,24 @@
 
 use Symfony\Component\Routing\Annotation\Route;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use App\Service\ProjectsData;
 
 class HomeController extends Controller
 {
     /**
      * @Route("/", name="home")
+     *
+     * @param ProjectsData $projects
+     * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function index()
+    public function index(ProjectsData $projects)
     {
+        $webPath = $this->get('kernel')->getProjectDir() . '/public/';
```

Comment:
`$this->get('kernel')` sintaksė buvo naudojama `Symfony 3`.
Su `Symfony 4` galima naudoti `KernelInterface` tiesiog funkcijos argumente.

Tokiu atveju kodas tampa daug lengviau testuojamas, perkeliamas į kitus projektus, service vis dar gali būti `public:false` ir daugiau yra įrankių kodo analizei (klaidų prevencijai).

Plačiau:
* https://symfony.com/doc/current/service_container.html
* https://api.symfony.com/2.8/Symfony/Component/HttpKernel/KernelInterface.html